Mathlib/Topology/ContinuousFunction/CFC.lean

```diff
@@ -0,0 +1,1111 @@
+generated by the element. -/
+theorem cfc₂_range_le : (cfc₂ 𝕜 a).range ≤ elementalStarAlgebra 𝕜 a := by
+  rintro _ ⟨f, rfl⟩
+  exact cfc₁_range_le 𝕜 a ⟨_, rfl⟩
+
+variable {𝕜}
+
+/-- The range of the continuous functional calculus is contained in the `elementralStarAlgebra`
+generated by the element. -/
+theorem cfc₁_mem_elementalStarAlgebra (f : C(spectrum 𝕜 a, 𝕜)) :
+    cfc₁ 𝕜 a f ∈ elementalStarAlgebra 𝕜 a :=
+  cfc₁_range_le 𝕜 a ⟨f, rfl⟩
+
+/-- The range of the continuous functional calculus is contained in the `elementalStarAlgebra`
+generated by the element. -/
+theorem cfc₂_mem_elementalStarAlgebra (f : C(𝕜, 𝕜)) : cfc₂ 𝕜 a f ∈ elementalStarAlgebra 𝕜 a :=
+  cfc₁_mem_elementalStarAlgebra _ _
+
+end Core
+
+section Normed
+
+variable (𝕜 : Type*) {A : Type*} [IsROrC 𝕜] [NormedRing A] [StarRing A] [NormedAlgebra 𝕜 A]
+    [StarModule 𝕜 A] [NormedStarGroup A] (a : A) [CompactSpace (spectrum 𝕜 a)] [CFCClass 𝕜 a]
+
+/-- For a continuous functional calculus for `a` over `IsROrC 𝕜` into a normed star ring, the
+range is precisely the `elementalStarAlgebra` generated by `a`. -/
+theorem cfc₁_range : (cfc₁ 𝕜 a).range = elementalStarAlgebra 𝕜 a := by
+  rw [StarAlgHom.range_eq_map_top, ← polynomialFunctions.starClosure_topologicalClosure, ←
+    StarSubalgebra.topologicalClosure_map _ _ (cfc₁_closedEmbedding 𝕜 a),
+    polynomialFunctions.starClosure_eq_adjoin_X, StarAlgHom.map_adjoin]
+  congr
+  rw [Set.image_singleton, toContinuousMapOnAlgHom_apply, cfc₁_map_X]
+
+-- this is the only direct result where we need the `topology.tietze_extension`
+-- and also `topology.metric_space.emetric_paracompact` for `normal_space` instance.
+theorem cfc₂_range : (cfc₂ 𝕜 a).range = elementalStarAlgebra 𝕜 a := by
+  refine' le_antisymm (cfc₂_range_le 𝕜 a) _
+  rw [← cfc₁_range]
+  rintro - ⟨f, rfl⟩
+  have hspec := (isCompact_iff_compactSpace.mpr (‹_› : CompactSpace (spectrum 𝕜 a))).isClosed
+  obtain ⟨f_re', hf_re⟩ :=
+    (ContinuousMap.comp ⟨IsROrC.re, IsROrC.continuous_re⟩ f).exists_restrict_eq_of_closed hspec
+  obtain ⟨f_im', hf_im⟩ :=
+    (ContinuousMap.comp ⟨IsROrC.im, IsROrC.continuous_im⟩ f).exists_restrict_eq_of_closed hspec
+  refine'
+    ⟨(@IsROrC.ofRealClm 𝕜 _ : C(ℝ, 𝕜)).comp f_re' +
+        @IsROrC.I 𝕜 _ • (@IsROrC.ofRealClm 𝕜 _ : C(ℝ, 𝕜)).comp f_im',
+      _⟩
+  simp only [AlgHom.toRingHom_eq_coe, map_add, RingHom.coe_coe, StarAlgHom.coe_toAlgHom]
+  rw [cfc₂, StarAlgHom.coe_comp, Function.comp_apply, Function.comp_apply, ← map_add]
+  congr!
+  ext x
+  apply IsROrC.ext <;>
+    simp [ContinuousMap.compStarAlgHom', ContinuousMap.restrict, ContinuousMap.comp, (· ∘ ·),
+      ContinuousMap.smul_apply IsROrC.I]
+  · exact FunLike.congr_fun hf_re x
+  · rw [← IsROrC.I_im' (f x)]
+    congr! 1
+    exact FunLike.congr_fun hf_im x
+
+/-- For an isometric continuous functional calculus for `a` over `IsROrC 𝕜`, the range is
+precisely the `elementralStarAlgebra` generated by `a`. -/
```

Comment:
```suggestion
/-- For an isometric continuous functional calculus for `a` over `IsROrC 𝕜`, the range is
precisely the `elementalStarAlgebra` generated by `a`. -/
```
It's really annoying when a spelling error makes it into muscle memory :-)

Reply:
Sorry, this was the result of a bad search and replace.